lib/fabrik/VirtualHostInstance.js

```diff
@@ -0,0 +1,93 @@
+const _ = require('lodash');
+const Promise = require('bluebird');
+const BaseInstance = require('./BaseInstance');
+const bosh = require('../bosh');
+const utils = require('../utils');
+const mapper = require('./VirtualHostRelationMapper');
+const catalog = require('../models').catalog;
+
+class VirtualHostInstance extends BaseInstance {
+
+  constructor(guid, manager) {
+    super(guid, manager);
+    this.director = bosh.director;
+    this.mapper = mapper.VirtualHostRelationMapper;
+  }
+
+  get async() {
+    return false;
+  }
+
+  initialize() {
+    return this.getDeploymentNameForVirtualHostInstanceId(this.guid)
+      .then((deploymentName) => this.deploymentName = deploymentName);
+  }
+
+  create(params) {
+    const rabbitmqInstanceName = params.parameters.dedicated_rabbitmq_instance;
+    return this.cloudController.getMultiTenantServiceInstanceIdWithName(rabbitmqInstanceName, params.organization_guid, params.space_guid, 'rabbitmq')
```

Comment:
Why is the `getMultiTenantServiceInstanceIdWithName` method retrieving service plan & validating the service plan? Anyways you seem to have added the lifecycle 'multi-tenancy'. After fetching instance guid, you can directly invoke the BOSH agent & if no agent supports multi-tenancy it directly throws the error `Feature Not supported`. Most of the times we can avoid unnecessary call to CloudController.

Reply:
getMultiTenantServiceInstanceIdWithName method retrieving service plan & validating the service plan just to make sure:
(1) Service instance name passed to create virtual host is a 'rabbitmq' service instance only.
(2) rabbitmq instance supports multi-tenancy & it is not a dev plan service instance.

Without this check the calls with reach broker agents of instances other then rabbitmq also if wrong instance name is provided.
Hope this is fine?

Reply:
Without this check the calls with reach broker agents of instances other then rabbitmq also if wrong instance name is provided.
-- Agreed, but it avoids the extra call to cloud controller. Either ways there is anyways an additional network call in case of wrong instance names - in your case ur validating against CC & i am suggesting let that validation be part of getting the request executed by the agents. The benefit of the later (along with validation) ensures additional call to CC is not required.